lib/phoenix/router/route.ex

```diff
@@ -50,6 +50,20 @@ defmodule Phoenix.Router.Route do
       dispatch: build_dispatch(route, binding)}
   end
 
+  @doc """
+  Used by the binding macros built in maybe_binding, to URI decode params
+  embedded in paths.
+
+  Strings are decoded directly. Lists of strings decodes each string.
+  """
+  def uri_decode_param_value(value) do
```

Comment:
I am not too happy about this function being here and public. It is public as it's used by the macros created in ```maybe_binding```. Is there a better way of doing this, or a better place to put it?